File: 136/main.py
"""
Given an array of integers, every element appears twice except for one. Find that single one.
"""

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        for v in nums:
            result ^= v
        return result

ans = Solution()
print(ans.singleNumber([1, 3, 2, 2, 8, 9, 1, 3, 9]))

File: 136/main.go
package main

import "fmt"

func main() {
	fmt.Println(singleNumber([]int{5, 1, 2, 1, 2}))
}

func singleNumber(nums []int) int {
	result := 0
	for _, val := range nums {
		result ^= val
	}
	return result
}
